octopoes/tests/integration/test_unicode.py

```diff
@@ -0,0 +1,74 @@
+import os
+import time
+import uuid
+from datetime import datetime
+
+import pytest
+
+from octopoes.api.models import Declaration, Observation
+from octopoes.connector.octopoes import OctopoesAPIConnector
+from octopoes.models import DeclaredScanProfile, ScanLevel
+from octopoes.models.ooi.dns.zone import Hostname
+from octopoes.models.ooi.network import Network
+from octopoes.models.origin import OriginType
+
+if os.environ.get("CI") != "1":
+    pytest.skip("Needs XTDB multinode container.", allow_module_level=True)
+
+use_unicode = True
+names = ["🐱", "★.com", "🐈"] if use_unicode else ["cat", "xn--p3h.com", "boefje"]
+
+
+def test_unicode_network(octopoes_api_connector: OctopoesAPIConnector, valid_time: datetime):
+    network = Network(name=names[0])
+    octopoes_api_connector.save_declaration(
+        Declaration(
+            ooi=network,
+            valid_time=valid_time,
+        )
+    )
+
+    time.sleep(1)
+
+    assert octopoes_api_connector.list_objects(types={Network}, valid_time=valid_time).count == 1
```

Comment:
This isn't really a strong test, a better approach would be to assert that `list_objects()` returns the `Network` object (and only that) created earlier

Reply:
Addressed in 459cc54449115c378f7fdb8ed035f0a9a4f9c461